dsperse/src/cli/prove.py

```
@@ -174,59 +239,182 @@ def is_run_root_dir(p):
-        return
+    run_results_path = os.path.join(run_dir, "run_results.json")
+    if not os.path.exists(run_results_path):
+        print(f"{Fore.YELLOW}Warning: run_results.json not found in {run_dir}; a new one will be created/updated.{Style.RESET_ALL}")
 
-    metadata_path = os.path.join(run_root_dir, "metadata.json")
-    if not os.path.exists(metadata_path):
-        print(f"{Fore.RED}Error: metadata.json not found in {run_root_dir}{Style.RESET_ALL}")
+    # Determine data_path: prefer explicit --from archive, else metadata.source_path, else model_path/slices
+    try:
+        meta = Utils.load_run_metadata(Path(run_dir))
+    except Exception as e:
```

Ruff (0.14.5)

Do not catch blind exception: `Exception`

(BLE001)